commit a9a5e0bdc5a77a7c662ad4be0ad661f0b0d5e99d
Author: Geert Uytterhoeven <geert+renesas@glider.be>
Date:   Tue Jan 7 09:38:57 2025 +0100

    hardening: Document INIT_STACK_ALL_PATTERN behavior with GCC
    
    The help text for INIT_STACK_ALL_PATTERN documents the patterns used by
    Clang, but lacks documentation for GCC.
    
    Signed-off-by: Geert Uytterhoeven <geert+renesas@glider.be>
    Link: https://lore.kernel.org/r/293d29d6a0d1823165be97285c1bc73e90ee9db8.1736239070.git.geert+renesas@glider.be
    Signed-off-by: Kees Cook <kees@kernel.org>

diff --git a/security/Kconfig.hardening b/security/Kconfig.hardening
index c9d5ca3d8d08..b56e001e0c6a 100644
--- a/security/Kconfig.hardening
+++ b/security/Kconfig.hardening
@@ -127,6 +127,7 @@ choice
 		  repeating for all types and padding except float and double
 		  which use 0xFF repeating (-NaN). Clang on 32-bit uses 0xFF
 		  repeating for all types and padding.
+		  GCC uses 0xFE repeating for all types, and zero for padding.
 
 	config INIT_STACK_ALL_ZERO
 		bool "zero-init everything (strongest and safest)"